### PR TITLE
Fix KMT task for building system-probe

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This fixes the KMT task for building system-probe and sharing with the micro-vms.
It also upgrade test-infra-definitions to pull in [changes](https://github.com/DataDog/test-infra-definitions/pull/751) allowing to vary VMs over vcpus and/or memory

### CODE REVIEW

**Review by gjulianm — APPROVED** [review]

File: .gitlab/common/test_infra_version.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ variables:
   # and check the job creating the image to make sure you have the right SHA prefix
   TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX: ""
   # Make sure to update test-infra-definitions version in go.mod as well
-  TEST_INFRA_DEFINITIONS_BUILDIMAGES: e488596ad7f3
+  TEST_INFRA_DEFINITIONS_BUILDIMAGES: 0700b1afb854
```
Comment (pducolin) [inline .gitlab/common/test_infra_version.yml:7]: **💬 suggestion**
There is probably an automated PR that bumps it too, can you close it?

**Review by pducolin — APPROVED** [review]

File: tasks/kmt.py [hunk 2]
```diff
@@ -734,20 +736,28 @@ def build(
...
+    if not os.path.exists(system_probe_yaml):
+        raise Exit(f"file {system_probe_yaml} not found")
+
     for d in domains:
         d.copy(ctx, "./bin/system-probe", "/root")
-        d.copy(ctx, f"kmt-deps/{stack}/shared.tar", "/")
+        d.copy(ctx, shared_archive, "/")
         d.run_cmd(ctx, "tar xf /shared.tar -C /", verbose=verbose)
+        d.run_cmd(ctx, "mkdir /opt/datadog-agent/run")
+        d.run_cmd(ctx, "mkdir /etc/datadog-agent/")
```
Comment (CelianR) [inline tasks/kmt.py:759]: **🥜 nitpick:** ‏
```suggestion
        d.run_cmd(ctx, "mkdir /etc/datadog-agent")
```

**Review by CelianR — APPROVED** [review]

File: .gitlab/common/test_infra_version.yml (hunk 1 above)
Comment (usamasaqib) [reply .gitlab/common/test_infra_version.yml:7]: Yes I will. Thanks for letting me know

**Pull request merged**